src/sdk/config_fs.js

```diff
@@ -265,104 +440,222 @@ class ConfigFS {
      */
     async get_account_by_name(account_name, options = {}) {
         const account_path = this.get_account_path_by_name(account_name);
-        const account = await this.get_config_data(account_path, options);
+        let account = await this.get_identity_config_data(account_path, { ...options, silent_if_missing: true });
+        if (!account) {
+            const old_account_path = this._get_old_account_path_by_name(account_name);
+            account = await this.get_identity_config_data(old_account_path, { ...options, silent_if_missing: true });
+        }
+        if (!account && !options.silent_if_missing) {
+            const err = new Error(`Could not find account by name ${account_name}`);
+            err.code = 'ENOENT';
+            throw err;
+        }
         return account;
     }
 
     /**
-     * list_root_accounts returns the root accounts array exists under the config dir
-     * @returns {Promise<Dirent[]>} 
-     */
-    async list_root_accounts(options) {
-        return nb_native().fs.readdir(this.fs_context, this.accounts_dir_path);
+    * @param {fs.Dirent} entry
+    * @param {string} suffix
+    * @returns {boolean}
+    */
+    _is_config_file(entry, suffix) {
```

Comment:
I think this name might be confusing, it doesn't guarantee that the file blabla.json is indeed a config file.